packages/react-tabs/src/components/Tab/Tab.types.ts

```diff
@@ -39,19 +54,28 @@ export type TabProps = ComponentProps<Partial<TabSlots>> & TabCommons;
  * State used in rendering Tab
  */
 export type TabState = ComponentState<TabSlots> &
-  TabCommons & {
+  Omit<TabCommons, 'disabled'> &
+  Required<Pick<TabCommons, 'disabled'>> & {
     /**
      * A tab supports 'transparent' and 'subtle' appearance.
      */
     appearance?: string;
     /**
-     * If this tab is selected.
+     * A tab can have only an icon slot filled and no content.
+     */
+    iconOnly: boolean;
+    /**
+     * The current selected tab value.
      */
-    selected?: boolean;
+    selectedValue?: TabValue;
     /**
      * A tab can be either 'small' or 'medium' size.
      */
     size: 'small' | 'medium';
+    /**
+     * This tab's value;
+     */
+    value: TabValue;
```

Comment:
You're already extending from `TabCommons` so the `value` property should already be here
```suggestion
```